### PR TITLE
feat: add network ip families support

### PR DESCRIPTION
**Pull request by heww**
Add network IP families support so that we can deploy the harbor cluster in ipv4 only network.

Signed-off-by: He Weiwei <hweiwei@vmware.com>

### CODE REVIEW

**Review by bitsf — APPROVED** [review]

File: apis/goharbor.io/v1beta1/chartmuseum_types.go [hunk 1]
```diff
@@ -61,6 +61,9 @@ type ChartMuseumSpec struct {
 
 	// +kubebuilder:validation:Required
 	Chart ChartMuseumChartSpec `json:"chart"`
+
+	// +kubebuilder:validation:Optional
+	Network *harbormetav1.Network `json:"network,omitempty"`
```
Comment (steven-zou) [inline apis/goharbor.io/v1beta1/chartmuseum_types.go:66]: As all harbor components need that field, why not put it into the `harbormetav1.ComponentSpec `json:"inline"`

**Review by steven-zou — COMMENTED** [review]
check my comment?

@heww

File: apis/goharbor.io/v1beta1/chartmuseum_types.go (hunk 1 above)
Comment (heww) [reply apis/goharbor.io/v1beta1/chartmuseum_types.go:66]: The Harbor type includes a HarborComponentsSpec. If the network is added to harbormetav1.ComponentSpec, there will be many networks in the Harbor type. It's confusing.

**Review by steven-zou — APPROVED** [review]
LGTM

**Pull request merged**